pkg/controller/config/config.go

```diff
@@ -140,3 +144,52 @@ func findCycle(path []SecretLocation, nodes map[SecretLocation]bool, edges map[S
 	}
 	return nil, false
 }
+
+// Load loads and parses the config at path.
+func Load(configLocation string) (c *Configuration, err error) {
+	// we never want config loading to take down the controller
+	defer func() {
+		if r := recover(); r != nil {
+			c, err = nil, fmt.Errorf("panic loading config: %v", r)
+		}
+	}()
+	c, err = loadConfig(configLocation)
+	if err != nil {
+		return nil, err
+	}
+	if err := c.Validate(); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func loadConfig(configLocation string) (*Configuration, error) {
+	stat, err := os.Stat(configLocation)
+	if err != nil {
+		return nil, err
+	}
+
+	if stat.IsDir() {
+		return nil, fmt.Errorf("configLocation cannot be a dir - %s", configLocation)
+	}
```

Comment:
I wonder if the `Stat` check is even needed... if we attempt to `ReadFile` something taht does not exist or is a dir we'll get a reasonable error anyway, no?

Reply:
Removed.